src/GraphQL.Tests/Bugs/Bug1769.cs

```diff
@@ -0,0 +1,91 @@
+using GraphQL.Execution;
+using GraphQL.SystemTextJson;
+using GraphQL.Types;
+using GraphQL.Validation;
+using GraphQL.Validation.Complexity;
+using System;
+using System.Linq;
+using System.Numerics;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace GraphQL.Tests.Bugs
+{
+    // https://github.com/graphql-dotnet/graphql-dotnet/pulls/1769
+    public class Bug1769 : QueryTestBase<Bug1769Schema>
+    {
+        private void AssertQueryWithError(string query, string result, string message, int line, int column, object[] path, Exception exception = null, string code = null, string inputs = null)
+        {
+            var error = exception == null ? new ExecutionError(message) : new ExecutionError(message, exception);
+            if (line != 0) error.AddLocation(line, column);
+            error.Path = path;
+            if (code != null)
+                error.Code = code;
+            var expected = CreateQueryResult(result, new ExecutionErrors { error });
+            AssertQueryIgnoreErrors(query, expected, inputs?.ToInputs(), renderErrors: true, expectedErrorCount: 1);
+        }
+
+        [Fact]
+        public async Task DocumentExecuter_has_valid_options() {
+            var de = new DocumentExecuter();
+            var valid = await de.ExecuteAsync(new ExecutionOptions()
+            {
+                Query = "{test}",
+                Schema = Schema,
+            });
+            await Assert.ThrowsAsync<InvalidOperationException>(async () =>
+            {
+                await de.ExecuteAsync(new ExecutionOptions()
+                {
+                    Query = null,
+                    Schema = Schema,
+                });
+            });
+            await Assert.ThrowsAsync<InvalidOperationException>(async () =>
+            {
+                await de.ExecuteAsync(new ExecutionOptions()
+                {
+                    Query = "{test}",
+                    Schema = null,
+                });
+            });
+            await Assert.ThrowsAsync<InvalidOperationException>(async () =>
+            {
+                await de.ExecuteAsync(new ExecutionOptions()
+                {
+                    Query = "{test}",
+                    Schema = Schema,
+                    FieldMiddleware = null,
+                });
+            });
+        }
+
+        [Fact]
+        public void query_is_empty() => AssertQueryWithError("", null, "Unable to determine operation from query.", 0, 0, (object[])null);
```

Comment:
Please add test for `"\t    \t \r\n"` query.

Reply:
Done